hub/api/dataset.py

```diff
@@ -343,6 +345,14 @@ def ingest(
             if os.path.samefile(src, dest):
                 raise SamePathException(src)
 
+        if len(os.listdir(src)) < 1:
+            raise AutoCompressionError(src)
```

Comment:
instead of 1 error message for all cases, could you add a message argument to AutoCompressionError? In this line, it should say the directory was empty